src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/quickfix/RemoveAnnotationConflictQuickFix.java

```diff
@@ -72,44 +85,82 @@ public RemoveAnnotationConflictQuickFix(boolean generateOnlyOneCodeAction, Strin
+
+        String annotationsToRemove;
+        if (data.getExtendedDataEntry(ANNOTATION_TO_REMOVE) instanceof String) {
+            annotationsToRemove = (String) data.getExtendedDataEntry(ANNOTATION_TO_REMOVE);
+        } else {
+            annotationsToRemove = "";
+        }
+
+        String[] annotationToRemove = new String[0];
+        if (Arrays.stream(annotations).anyMatch(x -> x.contains(annotationsToRemove))) {
+            annotationToRemove = new String[]{annotationsToRemove};
+        }
+
+        String name = toResolve.getTitle();
+        if (data.getExtendedDataEntry(ANNOTATION_LIST) instanceof String[]) {
+            annotationToRemove = (String[]) data.getExtendedDataEntry(ANNOTATION_LIST);
+            name = getLabel(annotationToRemove);
+        }
+
+        PsiElement declaringNode = getBinding(context.getCoveredNode());
+        ChangeCorrectionProposal proposal = new DeleteAnnotationProposal(name, context.getSource().getCompilationUnit(),
+                context.getASTRoot(), parentType, 0, declaringNode, annotationToRemove);
+
+        try {
+            WorkspaceEdit we = context.convertToWorkspaceEdit(proposal);
+            toResolve.setEdit(we);
+        } catch (Exception e) {
+            LOGGER.log(Level.WARNING, "Unable to create workspace edit for code action to remove annotation", e);
+        }
+        return toResolve;
+    }
+
+    protected void removeAnnotations(Diagnostic diagnostic, JavaCodeActionContext context,
                                      List<CodeAction> codeActions) {
         if (generateOnlyOneCodeAction) {
-            removeAnnotation(diagnostic, context, parentType, codeActions, annotations);
+            removeAnnotation(diagnostic, context, codeActions, annotations);
         } else {
             for (String annotation : annotations) {
-                JavaCodeActionContext newContext = context.copy(); // each code action needs its own context
-                PsiElement selectedNode = getBinding(newContext.getCoveredNode());
-                removeAnnotation(diagnostic, newContext, selectedNode, codeActions, annotation);
+                removeAnnotation(diagnostic, context, codeActions, annotation);
             }
         }
     }
 
-    protected static void removeAnnotation(Diagnostic diagnostic, JavaCodeActionContext context, PsiElement parentType,
-                                           List<CodeAction> codeActions, String... annotations) {
+    protected void removeAnnotation(Diagnostic diagnostic, JavaCodeActionContext context,
+                                    List<CodeAction> codeActions, String... annotations) {
+        Map<String, Object> extendedData = new HashMap<>();
+        String annotationToRemove = annotations[0];
+        extendedData.put(ANNOTATION_TO_REMOVE, annotationToRemove);
+        removeAnnotation(diagnostic, context, codeActions, extendedData, annotations);
```

Comment:
Something seems wrong here. An array of annotations is passed to this method but we're ignoring all but the first one. In the case where only one code action is produced how could this possibly work? Shouldn't we be storing the entire list here in extended data? I see there's an ANNOTATION_LIST but it's only written in one of the child classes. Why do we need both of these? It seems like this could be simplified.

Reply:
You are right , I made the required changes based on your comment.